R/gefs.R

```diff
@@ -0,0 +1,224 @@
+#' 
+#' Fetches GEFS forecast data for every 6 hours out to 384 hours past selected date. GEFS
+#' is an ensemble of 21 models that can be summarized to estimate likelihoods of forecasts.
+#'  
+#' @importFrom tidyr gather
+#' @export
+#' 
+#' @param var the variable to get. Must be one of the variables listed in \code{gefs_variables()}.
+#' @param lat,lon the longitude. Will be converted to the nearest GEFS available
+#'  longitude. If lon is a list of vlaues, it must be a sequential list, and
+#'  data are returned for the number of longitudes in the list starting with
+#'  the maximum value and incrementing through the indexed values for the 
+#'  length of the list.
+#' @param date A date/string formatted as YYYYMMDD.
+#' @param forecast_time a string indicating which time of day UTC the forecast is from. Options are "0000", "0600", "1200", "1800".
+#' @param ens_idx sequential list of ensembles to fetch. Default is all 21. Note that the ensembles are labelled 0-20, so ens_idx=1:3 will return ensembles 0, 1, and 2.
+#' @param time_idx sequential list of time increments to return. List is the index
+#'   of times, which are in 6 hour increments. (e.g. c(1,2) fetches the 6 and 12 hour forecast.)
+#' @param dims (not implemented) indices for additional dimensions to be included between lat, lon, ens, and time.
+#' @param raw logical to indicate whether to return raw data matrix or reshaped data frame.
+#' @param ... additional parameters passed to \code{ncvar_get}.
+#' @return a list containing metadata and accompanying data frame of forecast 
+#'   values. If lat/lon are not specified, the $data is an unprocessed matrix.
+#'   
+#' @references \url{https://www.ncdc.noaa.gov/data-access/model-data/model-datasets/global-ensemble-forecast-system-gefs}
+#' 
+#' @author Nicholas Potter \email{potterzot@@gmail.com}
+#' @examples \dontrun{
+#' 
+#' #avialable latitudes and longitudes
+#' gefs_latitudes()
+#' gefs_longitudes()
+#' 
+#' #get a list of all gefs variables
+#' gefs_variables()
+#' 
+#' #All GEFS dimensions
+#' gefs_dimensions()
+#' 
+#' #values for a specific dimension
+#' gefs_dimension_values("height_above_ground")
+#' 
+#' #example location.
+#' lat = 46.28125
+#' lon = -116.2188
+#' 
+#' #Get forecast for a certain variable.
+#' forecast = gefs("Total_precipitation_surface_6_Hour_Accumulation_ens", lat, lon)
+#' 
+#' #Fetch a different date (available up to 10 days prior to today)
+#' forecast_yesterday_prec = gefs("Total_precipitation_surface_6_Hour_Accumulation_ens",
+#'                                lat, 
+#'                                lon,
+#'                                date=format(as.Date(Sys.time()) - 1, "%Y%m%d"))
+#' 
+#' #specific ensemble and times, for the 1800 forecast. 
+#' # here ensembles 1-3 (ensembles are numbered starting with 0)
+#' # and time for 2 days from today at 1800
+#' var = "Temperature_height_above_ground_ens"
+#' gefs(var, lat, lon, forecast_time = "1800", ens_idx=2:4, time_idx=1:8)
```

Comment:
Ah, yes, It's not yet 1800 UTC so it fails. I will rewrite to add a date that is the previous day so that it will work regardless of the hour that someone runs it.

If you try `forecast_time = "0600" for ex., it should work. Will push an update in a sec.